HelgaShiryaeva/assignment2/BinaryTree.java

```diff
@@ -0,0 +1,104 @@
+package assignment2;
+
+import java.util.HashSet;
+import java.util.Optional;
+import java.util.Set;
+
+/**
+ * This class represents a Binary tree data structure

```

Comment:
It would be good to also mention assumptions that this class makes, i.e. that values in tree are unique.